src/arrayMethodIndexOf.js

```diff
@@ -5,9 +5,18 @@
  *
  */
 function applyCustomIndexOf() {
-  [].__proto__.indexOf2 = function() {
-    // write code here
+  [].__proto__.indexOf2 = function(token, index = 0) {
+    if (index < 0) {
+      index = index + this.length;
```

Comment:
Do not mutate function arguments create another variable(ideally create constants and do not reassign them)